### ISSUE TITLE
Array of strings primitives shows error in Sanity Studio

### ISSUE THREAD

**Issue by bjornwang** [sanity-io/sanity#1357]
**Describe the bug**
When using an array of string primitives, `type: 'array', of: [{type: 'string'}]`, Sanity Studio shows an "**Unable to resolve type for item**" error.

![image](https://user-images.githubusercontent.com/5180175/59160526-20899500-8ad7-11e9-97b3-9ce5f1da5017.png)

**To Reproduce**
Below code provokes the error. This is copied directly from Sanity's [Array Type documentation page](https://www.sanity.io/docs/schema-types/array-type#example-array-of-strings).
```
    {
      title: 'Names',
      name: 'names',
      type: 'array',
      of: [{type: 'string'}]
    }
```

**Expected behavior**
As the code is copied directly from Sanity's documentation, the field definition is not expected to cause any errors. 

**Which versions of Sanity are you using?**
@sanity/cli                 0.140.17 (latest: 0.140.33)
@sanity/base                0.140.30 (latest: 0.140.33)
@sanity/components          0.140.30 (latest: 0.140.33)
@sanity/core                0.140.30 (latest: 0.140.32)
@sanity/default-layout      0.140.30 (latest: 0.140.33)
@sanity/default-login       0.140.15 (up to date)
@sanity/desk-tool           0.140.30 (latest: 0.140.33)
@sanity/google-maps-input   0.140.30 (latest: 0.140.33)
@sanity/production-preview   0.140.0 (up to date)
@sanity/vision              0.140.31 (latest: 0.140.33)

**What operating system are you using?**
Mac OS Sierra, 10.12.6

**Which versions of Node.js / npm are you running?**
6.5.0
v11.8.0

**Comment by TkachenkoDmitry** [comment]
Same with 'datetime':
` {
      name: 'dates',
      title: 'Dates',
      type: 'array',
      of: [
        {
          type: 'datetime'
        }
      ]
    }
}`

    "@sanity/base": "^0.142.0",
    "@sanity/components": "^0.142.0",
    "@sanity/core": "^0.142.0",
    "@sanity/default-layout": "^0.142.0",
    "@sanity/default-login": "^0.142.0",
    "@sanity/desk-tool": "^0.142.0"

Mac Os Mojave 10.14.6
Node: v10.15.3
NPM: 6.9.0

**Comment by rexxars** [comment]
Thanks for reporting! This has been fixed in #1540
